### ISSUE TITLE
Show subway entrance numbers

### ISSUE THREAD

**Issue by strump** [organicmaps/organicmaps#1778]
Each subway entrance/exit has number in Kyiv, Kharkiv, Minsk, Moscow, London. E.g. node [#1578323663](https://www.openstreetmap.org/node/1578323663) in Kyiv. 

OSM.org and Google Maps show entrance number:

![subway_entrance_osm](https://user-images.githubusercontent.com/720808/148075262-110014bf-80a2-419c-a280-bda4c0365b61.png)

![subway_entrance_google_map](https://user-images.githubusercontent.com/720808/148075274-8efebcc6-8c7e-443a-8247-87c7e912ae6b.png)

AFAIK, OrganicMaps shows station name under subway entrance icon. Is it possible to rewrite map style to show station name along with entrance number?

**Comment by starsep** [comment]
I think the issue is in OSM data.
According to 
https://wiki.openstreetmap.org/wiki/Tag:railway=subway%20entrance?uselang=en
> (name) **Not** the name of the station (use the public_transport=stop_area relation instead to link with the station). This doesn't include the reference either. 

Example subway entrances with ref without name. Warsaw, Metro Kabaty
![Screenshot_2022-01-04-17-20-30-41_75d0290a7c00525acd83c0ab8700d64c](https://user-images.githubusercontent.com/2798728/148090186-d4f825db-f614-4de9-9124-41d169de36f8.jpg)

**Comment by strump** [comment]
So subway entrance in Poland [#279152171](https://www.openstreetmap.org/node/279152171) has tag `ref=4` which is shown in OrganicMaps. While subway entrance in Kyiv [#1578323663](https://www.openstreetmap.org/node/1578323663) has tag `ref=4` and `name=...` and only name is visible.

So I can fix Kyiv's subway by removing `name` tag, but what is better? Should map display subway name near entrance point or entrance number?

**Comment by starsep** [comment]
I think the current rendering (due to tagging) in Warsaw makes sense: `ref` for entrances and `name` for stations. Alternatively both `name` and `ref` could be rendered for entrances.

I can think of two cases of name different than station name which make sense to me:
1. It is exit or entry only - like many entrances in Rome. I don't see any better tag for this information right now.
2. If it is overground interchange to another subway line/train station.

Ideally both these cases would be tagged somehow differently but putting such information in name could be the easiest solution.

**Comment by biodranik** [comment]
Wait, why numbers are needed at all? How people are using them? I hear for the first time about entrance numbers and didn't see them in real-life neither in Kyiv nor in Minsk.

**Comment by starsep** [comment]
In Warsaw there are information boards with these numbers
![Tablica SIP jpg_900-528](https://user-images.githubusercontent.com/2798728/148121008-637684a8-6d25-4241-b92c-6b9888299b62.jpg)

The above is on the platform, there are more next to entrances with just a single number.

It is easier to exit from the right side or to meet at the particular entrance.

**Comment by pastk** [comment]
Yeap, in Moscow subway exits are numbered also and they're used in navigational maps and signs inside of stations.

**Comment by strump** [comment]
Here's sample of exit numbers in Kyiv:

![2022-01-10 10 18 47](https://user-images.githubusercontent.com/720808/148745513-49993983-7f3c-4da1-a7ad-26a91070e979.jpg)

**Comment by pastk** [comment]
Ideally the map should display refs and when an entrance is selected there should be a name displayed in the bottom sheet too, I think.

**Comment by olivertzeng** [comment]
Would really hope that a fix is implemented for this issue, any updates?

**Comment by map-per** [comment]
> I think the issue is in OSM data.
> According to
> https://wiki.openstreetmap.org/wiki/Tag:railway=subway%20entrance?uselang=en
>
> > (name) Not the name of the station (use the public_transport=stop_area relation instead to link with the station). This doesn't include the reference either.

I agree, the names in Kyiv should be removed form OSM as they are the name of the station and not the name of the entrance.

**Comment by brunetton** [comment]
To summarize:

- when `name` is given, OrganicMaps only shows the entrance name
- when `name` is not given, OrganicMaps shows the entrance number (`ref`, if present)

As proposed by @map-per:

> Ideally the map should display refs and when an entrance is selected there should be a name displayed in the bottom sheet too, I think.

I also think this is the way to go. Refs take up much less space on the map than names, and are really useful. And I would add:
- also display subway station name (the subway stop to which the exit belongs) in the bottom sheet. But it implies to have the relation in data, not sure this is the case. Example: https://www.openstreetmap.org/relation/9461459

## Examples

### Warsaw: no name -> exits numbers are displayed

As stated [here](https://github.com/organicmaps/organicmaps/issues/1778#issuecomment-1005036742)

https://www.openstreetmap.org/node/279152171 (geo: 52.131367,21.065938)

| key | value |
|--------|--------|
| railway | subway_entrance |
| ref     | 4               |
| wheelchair | no           |

![Image](https://github.com/user-attachments/assets/22e6ce51-3c8d-4ada-855a-d5c71ca4c131)

### Paris: name present -> exists numbers are **not** displayed

https://www.openstreetmap.org/node/674728407 (geo: 48.898064,2.358937)

| key       | value              |
|-----------|--------------------|
| railway   | subway_entrance    |
| ref       | 1                  |
| name      | Boulevard Ney      |

![Image](https://github.com/user-attachments/assets/149c628e-92a6-451a-9fdc-9464f90f2dd3)

### Vienna: name present -> exists numbers are **not** displayed

https://www.openstreetmap.org/node/1668557470 (geo: 48.2194524,16.3914159)

| key        | value           |
|------------|-----------------|
| railway    | subway_entrance |
| name       | Nordbahnstraße  |
| ref        | U1,U2           |
| wheelchair | no              |


![Image](https://github.com/user-attachments/assets/29a76b88-56ea-43c2-ade4-c61e1652a149)
